src/main/java/com/conveyal/datatools/manager/jobs/FeedToMerge.java

```diff
@@ -0,0 +1,40 @@
+package com.conveyal.datatools.manager.jobs;
+
+import com.conveyal.datatools.manager.models.FeedVersion;
+import com.conveyal.gtfs.loader.Table;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.SetMultimap;
+import com.google.common.collect.Sets;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.zip.ZipFile;
+
+import static com.conveyal.datatools.manager.utils.MergeFeedUtils.getIdsForTable;
+
+/**
+ * Helper class that collects the feed version and its zip file. Note: this class helps with sorting versions to
+ * merge in a list collection.
+ */
+public class FeedToMerge {
+    public FeedVersion version;
+    public ZipFile zipFile;
+    public SetMultimap<Table, String> idsForTable = HashMultimap.create();
+    public Set<String> serviceIds = new HashSet<>();
+    private static final Set<Table> tablesToCheck = Sets.newHashSet(Table.TRIPS, Table.CALENDAR, Table.CALENDAR_DATES);
+
+    public FeedToMerge(FeedVersion version) throws IOException {
+        this.version = version;
+        this.zipFile = new ZipFile(version.retrieveGtfsFile());
```

Comment:
At some point, it is critical for this zipFile to be closed. In looking at the source code, it looks like the ZipFile class can have a bunch of streams operating on it at once. And it seems that whenever these gtfs-lib csvReaders are created a new stream is created. Once the `zipFile.close()` method is called, all of these streams are then closed. So, somewhere in the code in a finally block after all reading from this ZipFile is finished, this must be closed.

Reply:
Partially handled in https://github.com/ibi-group/datatools-server/pull/376/commits/e90b8c1e126267f0984a139725010523284e9136.

Reply:
Done handling in https://github.com/ibi-group/datatools-server/pull/376/commits/8179f873f950f83394f7c628131f9b2e73fc72e1.